server/src/main/java/org/elasticsearch/index/codec/vectors/VectorScoringUtils.java

```diff
@@ -34,40 +36,74 @@ private VectorScoringUtils() {}
      * @throws IOException if an I/O error occurs
      */
     public static void scoreAndCollectAll(KnnCollector knnCollector, AcceptDocs acceptDocs, RandomVectorScorer scorer) throws IOException {
```

Comment:
This should just use the "vectorvalues" interface provided higher up. This provides an iterator.